Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -1,13 +1,201 @@
 import Foundation
 import mongoc
 
+/**
```

Comment:
hmm. I can see arguments for both. 
maybe it is strange to define them in a different file via extension when they are used throughout the main file. along those lines, maybe it would make sense to move the initializers that use CodingStrategies into extensions in the new file....
I don't feel very strongly either way though, so I'll defer to you. mainly I just wanted to avoid making the encoder file any longer than it already is.

Reply:
I think the strategy definitions should stay in the respective encoder/decoder files for the reasons you mention. 

Because one of the initializers is part of the public API, I think it's better that they both stay in the respective files as well. That way someone browsing the code can easily see all the possible ways to instantiate them. 

That said, if the files continue to grow, we can evaluate later whether or not its worth splitting them up into extensions, similar to what we did with `Document`.